tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php

```diff
@@ -334,6 +477,39 @@ public function test_that_draft_posts_that_are_posted_within_the_last_x_days_can
 		$this->assertEquals( 10, count( $trash_posts ) );
 	}
 
+	/**
+	 * Test date filter (within the last x days) with single post status.
+	 */
+	public function test_that_draft_posts_that_are_posted_within_the_last_x_days_can_be_deleted() {
+		$date = date( 'Y-m-d H:i:s', strtotime( '-3 day' ) );
+
+		$draft_posts = $this->factory->post->create_many( 10, array(
+			'post_type'   => 'post',
+			'post_status' => 'draft',
+			'post_date'   => $date,
+		) );
+
+		$this->assertEquals( 10, count( $draft_posts ) );
+
+		$delete_options = array(
+			'post_status'  => array( 'draft' ),
+			'limit_to'     => -1,
+			'restrict'     => true,
+			'force_delete' => true,
+			'date_op'      => 'after',
+			'days'         => '5',
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
```

Comment:
Assert that some additional posts that don't match the condition are not deleted.

Do this for all the other test cases as well.

Reply:
Fixed